components/widgets/resources/src/main/resources/resources/uicomponents/widgets/extraGeneVariantData.js

```diff
@@ -149,12 +149,14 @@ var ExtraGeneVariantData = (function (ExtraGeneVariantData) {
       var buttonRowInner = this.addVariantButtonTemplate.evaluate(data);
       var geneRowInner = this.geneRowTemplate.evaluate(data);
 
-      geneRowInner = geneRowInner.replace(/_\d+_|=\d+&/g, function (placeholder) {
+      geneRowInner = geneRowInner.replace(/_\d+_|=\d+&|xwiki-form-strategy-\d+/g, function (placeholder) {
```

Comment:
This works, however I have 2 comments:
- while I love regular expressions, I'm not happy with using them for manipulating html fragments because they can so easily produce invalid results;
- the property names are hard coded; if we add more fields displayed as checkboxes (or radio buttons), we need to add even more javascript code to fix those.

I wonder if this can be done in a safer way (using dom manipulation instead of regex, e.g. _after_ the fragment is inserted in the document) and using a more generic approach that would capture all cases (for instance, they're all inputs in labels with the css class "xwiki-form-listclass"). It would suffice to remove the label's for attribute and the input's id attribute, no need to actually replace the values. 
@veronikaslc Thoughts?